azurelinuxagent/pa/provision/factory.py

```diff
@@ -23,15 +23,23 @@
 from .default import ProvisionHandler
 from .cloudinit import CloudInitProvisionHandler, cloud_init_is_enabled
 
-def get_provision_handler(distro_name=DISTRO_NAME,  # pylint: disable=W0613
-                            distro_version=DISTRO_VERSION,  # pylint: disable=W0613
-                            distro_full_name=DISTRO_FULL_NAME):  # pylint: disable=W0613
 
+def cloud_init_should_provision():
```

Comment:
I have concerns about changing files in azurelinuxagent/pa

Thoughts on if it would be better to copy this logic somewhere else, and leave this file as is?